packages/block-library/src/cover/shared.js

```diff
@@ -109,3 +117,62 @@ export function getPositionClassName( contentPosition ) {
 
 	return POSITION_CLASSNAMES[ contentPosition ];
 }
+
+function retrieveFastAverageColor() {
+	if ( ! retrieveFastAverageColor.fastAverageColor ) {
+		retrieveFastAverageColor.fastAverageColor = new FastAverageColor();
+	}
+	return retrieveFastAverageColor.fastAverageColor;
+}
+
+/**
+ * getCoverIsDark is a method that specifyies if the cover background is dark or not and
+ * applies the relevant attribute to help ensure that text is visible by default.
+ * This needs to be recalculated in all of the following Cover block scenarios:
+ * - When an overlay image is added, changed or removed
+ * - When the featured image is selected as the overlay image, or removed from the overlay
+ * - When the overlay color is changed
+ * - When the overlay color is removed
+ * - When the dimRatio is changed
+ *
+ * See the comments below for more details about which aspects take priority when
+ * calculating the relative darkness of the Cover.
+ *
+ * @param {string} url
+ * @param {number} dimRatio
+ * @param {string} overlayColor
+ * @return {boolean|Promise} True if cover should be considered to be dark.
```

Comment:
Does this ever return a promise? Now that there's an `await` on `retrieveFastAverageColor()` it looks like it always returns a boolean.

If that's the case, then I _think_ the code in `edit/index.js` might not need to be async functions, since the await is handled internally by `getCoverIsDark`? I could be wrong, though, so apologies if that's a red herring!

Reply:
Nevermind, thanks for explaining how `async` works in this case @glendaviesnz! For reference in case anyone else looking at this made the same mistake as me here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#description